### ISSUE TITLE
One var rule giving indentation errors

### ISSUE THREAD

**Issue by alallier** [eslint/eslint#3210]
ESLint version `1.0.0-rc-3`

Code example:
```
var test = something,
	test2 = something,
	test3;
```

Rule:
`"one-var": [2, "always"]`

ESLint Output:
```
2  error  Expected indentation of 4 characters  indent
3  error  Expected indentation of 4 characters  indent
```

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by gyandeeps** [comment]
Duplicate https://github.com/eslint/eslint/issues/3164
Rule is working as expected.

**Comment by alallier** [comment]
`"indent": [2, 2]` Although changing this to 4 would require every line in the js code to have an indentation of 4. The one var rule should have an indentation of 4 with the rest of the js code should have an indentation 2. This was working correctly 0.24.1

**Comment by gyandeeps** [comment]
In 0.24.1, we never accounted for this scenario for indent rule. Similar concern as #3164 .
You want all of your code to have 2 indent but here you want 4 indent. Which is not correct.

**Comment by alallier** [comment]
Yes that is what I want

**Comment by kethinov** [comment]
The idea behind it is this:

```js
// my hello world application
var x = 5;

function blah() {
  var y = 5,
      z = 2,
      a = 6;

  console.log(y);
  if (z > 1) {
    console.log(z);
  }
  return a;
}

blah();
```

That is a common way of doing indenting and there should be a config that will allow it to pass eslint.

4 spaces for var lists, 2 spaces for everything else.

**Comment by gyandeeps** [comment]
Here is when its gets dirty, since right now you are suing `var` thats why you want 4 but what would you do when you use `const`?

**Comment by kethinov** [comment]
I see nothing wrong having a rule that also permits this to be valid:

```js
// my hello world application
var x = 5;

function blah() {
  const p,
        q,
        r;

  var y = 5,
      z = 2,
      a = 6;

  console.log(y);
  if (z > 1) {
    console.log(z);
  }
  return a;
}

blah();
```

People commonly want to comma separate their var/const/whatever lists with newlines and want it line up with the first item on the list, just as @targos from #3164 did. If eslint 1.x.x sees a stable release with this change that many would see as a regression, I suspect we'll see a lot more issues like this filed.

**Comment by gyandeeps** [comment]
@eslint/eslint-team Suggestions/views?

**Comment by IanVS** [comment]
It seems like a reasonable request to me. It does look a lot better than:

```
var foo,
  bar,
  baz
```

**Comment by ilyavolodin** [comment]
I think it's actually a valid request, it's just phrased incorrectly. For `var` it's not indentation, it's actually an alignment. You are aligning based on variable names. If you view a request under that light, I think it's a perfectly valid approach.

**Comment by kethinov** [comment]
Ah yes, you're right. Alignment is definitely a much better way to describe this.

**Comment by nzakas** [comment]
You can slice and dice semantics, but at the end of the day, this is indentation (with the purpose of alignment).

I think this is one of those things were we need to provides options to increase or decrease indentation level based on some AST node. So maybe instead of `indentSwitchCase: true`, we should have `SwitchCase: 1` (default of 0) and then in this case we'd have `VariableDeclarator: 2` (default of 1).

**Comment by ilyavolodin** [comment]
@nzakas That sounds reasonable, and should cover us for other scenarios as well. (like `const`)

**Comment by nzakas** [comment]
Let's defer on `const` for now and focus on everything else.

**Comment by gyandeeps** [comment]
working on this.

**Comment by estk** [comment]
The const behavior is kinda weird. Is a fix planned?

**Comment by Igloczek** [comment]
How about `const` weird behaviour?
